custom_components/localtuya/pytuya/__init__.py

```diff
@@ -457,12 +461,12 @@ async def detect_available_dps(self):
         self.log.debug("detected dps: %s", detected_dps)
         return detected_dps
 
-    def add_dps_to_request(self, dps_index):
+    def add_dps_to_request(self, dp_index):
```

Comment:
Here I would leave dps_index (or dp_indices), since this can actually be a list of DPs...